File: Data Structures & Algorithms/non-overlapping-intervals/submission-12.cs
public class Solution {
    public int EraseOverlapIntervals(int[][] intervals) {
        Array.Sort(intervals, (a, b) => a[0].CompareTo(b[0]));
        return intervals.Length - Dfs(intervals, 0, -1);
    }

    private int Dfs(int[][] intervals, int i, int prev) {
        if (i == intervals.Length) return 0;
        int res = Dfs(intervals, i + 1, prev);
        if (prev == -1 || intervals[prev][1] <= intervals[i][0]) {
            res = Math.Max(res, 1 + Dfs(intervals, i + 1, i));
        }
        return res;
    }
}

File: Data Structures & Algorithms/count-subsequences/submission-12.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {number}
     */
    numDistinct(s, t) {
        if (t.length > s.length) {
            return 0;
        }
        
        const dfs = (i, j) => {
            if (j === t.length) {
                return 1;
            }
            if (i === s.length) {
                return 0;
            }

            let res = dfs(i + 1, j);
            if (s[i] === t[j]) {
                res += dfs(i + 1, j + 1);
            }
            return res;
        }

        return dfs(0, 0);
    }
}